app/src/main/java/com/jerboa/feat/BlurNsfwTypes.kt

```diff
@@ -0,0 +1,27 @@
+package com.jerboa.feat
+
+import android.os.Parcelable
+import androidx.annotation.StringRes
+import com.jerboa.R
+import com.jerboa.datatypes.types.PostView
+import kotlinx.parcelize.Parcelize
+
+@Parcelize
+enum class BlurNsfwTypes(@StringRes val resId: Int) : Parcelable {
+    Nothing(R.string.app_settings_nothing),
+    All(R.string.account_settings_all),
+    OnlyFromNsfwCommunity(R.string.app_settings_blur_only_from_nsfw_community),
+    OnlyFromSfwCommunity(R.string.app_settings_blur_only_from_sfw_community),
+}
+
+fun BlurNsfwTypes.needNsfwBlur(postView: PostView) =
+    this.needNsfwBlur(postView.community.nsfw, postView.post.nsfw)
+
+fun BlurNsfwTypes.needNsfwBlur(isCommunityNsfw: Boolean, isPostNsfw: Boolean = isCommunityNsfw): Boolean {
+    return isPostNsfw && when (this) {
```

Comment:
The logic here doesn't follow. We already have a `blur_nsfw` setting in the back-end. And that should blur any NSFW post or NSFW community posts. I don't see why other options would be necessary.

The *only* other case I could think of, would be  a third option which is to blur everything, but that should probably be handled in the back end.

Reply:
This PR was made to solve 
https://github.com/dessalines/jerboa/issues/957

Which wants a more finegrained control over which images get blurred

Reply:
But I don't think this PR fully solves that FR, as that one wants to only not blur while it is in a communty mode view and still blur NSFW images that are in the feed

Reply:
Also this option does exists in other Reddit clients like IfR

Reply:
We should at least get the options figured out. The ones I can make sense of

- Blur everything
- Blur nothing
- Blur NSFW
- Blur NSFW except from NSFW communities ( I think that handles #957 , but I can't really tell)

Reply:
If we are doing Blur everything then, the title for the option should change from `Blur NSFW images` to `Blur images`

@MakcNmyc What do you think of this direction since its your PR

Reply:
Honestly, I don't catch idea why anyone would want to blur all posts from their feed. When I started working on the issue, I thought that we only needed to add an option for those who want to blur too explicit content (in NSFW communities), but save not quite hot (outside NSFW communities, but still NSFW). 
But I just started working on this project, I just started learning how to work in the GitHub open source community. And I’m ready to accept and correct the commit, no matter what you choose.

Reply:
`Blur everything` could be covered by #1239 where you can disable loading all images instead of just blurring them all instead

I would prefer that over extending the scope of this feature from NSFW to all images. What do you think @dessalines ?

Then we are back to 3 options

Reply:
That sounds good to me.

Reply:
@MakcNmyc Do you want to make that change? or else I'll do it

Reply:
> But I don't think this PR fully solves that FR, as that one wants to only not blur while it is in a communty mode view and still blur NSFW images that are in the feed

@MV-GH  Maybe fix it to work this way?

Reply:
> @MakcNmyc Do you want to make that change? or else I'll do it

I will do this if you explain what exactly needs to be done. Do you need the options to work for all images (not just nsfw)? But how will “NSFW Blur other than NSFW Communities” work?

Reply:
Address my comments above, but also make sure there are only 3 enum types, and that they make logical sense (explained above).

Reply:
I did the second option with some changes